DancingGoat/Utils/DomainBasedDeliveryClientFactory.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Kentico.Kontent.Delivery.Abstractions;
+using Kentico.Kontent.Delivery.Builders.DeliveryClient;
+
+namespace DancingGoat.Utils
+{
+    public class DomainBasedDeliveryClientFactory : IDeliveryClientFactory
+    {
+        private readonly IDeliveryClientFactory _baseClientFactory;
+        private readonly ProjectContext _projectContext;
+        private readonly ITypeProvider _typeProvider;
+        private readonly IContentLinkUrlResolver _linkResolver;
+
+        public DomainBasedDeliveryClientFactory(
+            IDeliveryClientFactory baseClientFactory,
+            ProjectContext projectContext,
+            ITypeProvider typeProvider,
+            IContentLinkUrlResolver linkResolver
+        )
+        {
+            _baseClientFactory = baseClientFactory;
+            _projectContext = projectContext;
+            _typeProvider = typeProvider;
+            _linkResolver = linkResolver;
+        }
+
+        public IDeliveryClient Get()
+        {
+            var clientBuilder = _projectContext.IsInPreviewMode
+                ? DeliveryClientBuilder.WithOptions(builder => builder
+                    .WithProjectId(_projectContext.ProjectGuid)
+                    .UsePreviewApi(_projectContext.PreviewApiKey)
+                    .Build()
+                )
+                : DeliveryClientBuilder.WithOptions(builder => builder
+                    .WithProjectId(_projectContext.ProjectGuid)
+                    .UseProductionApi()
+                    .Build()
```

Comment:
There are two `.Build()` calls, one inside the lambda and one outside. I did it this way because it felt better to have redundant code than mixing the ternary in the middle of it. This approach gives clear view of the two options that may apply

Reply:
I refactored it slightly in the end.